composer/packages/vs-syntax-tree/src/components/dropdown/dropdownNode.tsx

```diff
@@ -0,0 +1,159 @@
+"use strict";
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+import React, { useEffect, useState } from "react";
+import { Icon } from "semantic-ui-react";
+import { COLLAPSED_ARROW_ICON,
+         COLLAPSIBLE_ARROW_ICON,
+         DEFAULT_DROPDOWN_COLOR,
+         DROPDOWN_LOCATE_ICON,
+         DROPDOWN_WARNING_ICON,
+         LARGE_ICON,
+         NON_COLLAPSIBLE_ICON,
+         PRIMARY_COLOR,
+         SMALL_ICON,
+         WARNING_COLOR } from "../../resources/constants";
+import { DropdownNodeProps } from "../../resources/tree-interfaces";
+import * as styles from "../../styles/dropdown-tree.styles";
+
+function DropdownNode(props: DropdownNodeProps) {
+    const [ifCollapsible, setIfCollapsible] = useState(false);
+    const [isCollapsed, setIsCollapsed] = useState(false);
+    const [showDetails, setShowDetails] = useState(false);
+    const [hoverStatus, setHoverStatus] = useState(false);
+
```

Comment:
Let's destructure props and use it within the component. This is a good example where destructuring is useful.